### ISSUE TITLE
[YSQL] Title

### ISSUE THREAD

**Issue by suppersam1** [yugabyte/yugabyte-db#19699]
### Description

Why is ybbasectid the key column of an index table, and is its query efficiency consistent with the efficiency of using ybctid to search a table?

```json
table_name: "test1_embedding_idx"
schema {
  columns {
    id: 10
    name: "embedding"
    type {
      main: BINARY
    }
    is_key: true
    is_hash_key: true
    is_nullable: false
    is_static: false
    is_counter: false
    sorting_type: 0
    order: 1
    pg_type_oid: 0
  }
  columns {
    id: 11
    name: "ybidxbasectid"
    type {
      main: BINARY
    }
    is_key: true
    is_nullable: false
    is_static: false
    is_counter: false
    sorting_type: 1
    order: -101
    pg_type_oid: 0
  }
}
```

### Issue Type

kind/question

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

**Comment by ddorian** [comment]
What's the schema & index above in SQL terms?

And which query do you mean (along with an explain plan) ?

**Comment by FranckPachot** [comment]
In a non unique secondary index the ybbasectid (which is the ybctid of the table, the internal representation of its primary key) must be part of the key to have unique entries. It is at the end of the indexed columns so doesn't impact the range scan. And it is used to access the table's row to get the columns that are not in the index (in an Index Scan that is not an Index Only Scan)

**Comment by suppersam1** [comment]
> In a non unique secondary index the ybbasectid (which is the ybctid of the table, the internal representation of its primary key) must be part of the key to have unique entries. It is at the end of the indexed columns so doesn't impact the range scan. And it is used to access the table's row to get the columns that are not in the index (in an Index Scan that is not an Index Only Scan)

 am currently trying to develop a plugin that requires updating one row of data in the index table. Can I use ybbasectid instead of index fields as the where condition (as the length of index fields may be large)? I tried using BMCpgnewselect and binding ybbasectid as the where condition to query the corresponding index data rows, it's ok. But when I used the same method to update, it would prompt me that the update must be bound to ybctid or primary key.

And I want to know if using ybbasectid as the where condition to filter a certain row in the index table would be very efficient.

**Comment by FranckPachot** [comment]
Can you explain more about this plug-in? I'm curious. 

I've no idea what is BMCpgnewselect and how you access those internal columns but basically accessing with ybbasectid and without the index key columns cannot be efficient as it is a suffix.

**Comment by suppersam1** [comment]
> Can you explain more about this plug-in? I'm curious.
> 
> I've no idea what is BMCpgnewselect and how you access those internal columns but basically accessing with ybbasectid and without the index key columns cannot be efficient as it is a suffix.

In my current understanding, the index table of yb is the same concept as a common table. So I want to use index tables like a common table.However, the index table of pg is composed of index columns and containing columns, and users cannot flexibly customize the columns. This makes it difficult to introduce yb into some complex indexes, such as hnsw and ivf of pgvector. But if we can customize the columns of the index table, we can easily introduce these indexes. In fact, I am trying to do these things. In DefineIndex(), I forcefully constructed the structure of the hnsw index table using the if statement, and then stored the binary data of hnsw index.

**Comment by FranckPachot** [comment]
Yes, indexes are like tables, with the addition of this `ybidxbasectid` that references the table's row key.
What you are doing looks cool. I'll ping the colleagues who have been working on the YugabyteDB implementation of GIN indexes and who are looking at indexing for pgvector.  

We have also a #contributors slack channel at https://join.slack.com/t/yugabyte-db/shared_invite/zt-xbd652e9-3tN0N7UG0eLpsace4t1d2A

**Comment by ddorian** [comment]
Closing. Please reopen the issue if you have further questions.
